### PR TITLE
fix(snowflake): ensure that temp tables are only created once

### PR DESCRIPTION
**Pull request by cpcloud**
This PR ensures that we only create temp tables from dataframes in snowflake once.

### CODE REVIEW

**Comment by cpcloud** [comment]
Snowflake backend tests look good:

```
…/ibis on  fix-double-create-temp-table is 📦 v5.1.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env) took 6s
❯ pytest -m snowflake -n auto --dist loadgroup --snapshot-update --allow-snapshot-deletion -q
bringing up nodes...
....xxxxx.xxx....x......x................xxx..x.xx.....x.....x.s......x.........xx..x....xxx.....xxxx..x...xx.....x....x....x.....x.x.....x....................x...x.x.xx.....x....xx..........x......x....x...x..... [ 23%]
.......x.............x.............................xx.......x.x...x............x.........xxxxx......x...x.....x.....x..xx....x..........xx.xxx.x...x....x..x...x...x...x.x....x........x..xx........x..x.x..xxxx..... [ 46%]
.....x.xx.....x..x..............x............x....x.x.......................x...x..x....x......x...x..............sx.....x..x...s....x.................x...........x....................x...................x..s..x.. [ 70%]
.x.......x...............s.x.....................x.......xx...xx.x.xxxxxxxxxxxx...xxx.xx........................x..........x..............................x....................x....x.......x..........x............. [ 93%]
..............................xs...x.....s...x.........                                                                                                                                                               [100%]
746 passed, 7 skipped, 154 xfailed in 99.21s (0:01:39)
```

File: ibis/backends/snowflake/__init__.py [hunk 1]
```diff
@@ -406,20 +406,22 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
...
-                f"""
-                COPY INTO {table} ({column_names})
-                FROM (SELECT {parquet_column_names} FROM @{stage})
-                FILE_FORMAT=(TYPE=PARQUET COMPRESSION=AUTO)
-                PURGE=TRUE
-                """
-            )
+            if con.exec_driver_sql(f"SHOW TABLES LIKE '{table}'").scalar() is None:
+                con.exec_driver_sql(
+                    f"CREATE TEMP TABLE IF NOT EXISTS {table} ({schema})"
```
Comment (jcrist) [inline ibis/backends/snowflake/__init__.py:411]: IIUC the `if` statement above is checking if the table already exists before running this block - if so, should we drop the `"IF NOT EXISTS"` here then?

**Review by jcrist — APPROVED** [review]
Overall seems fine, just one question.

File: ibis/backends/snowflake/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/snowflake/__init__.py:411]: Yeah, thanks for catching!

**Comment by cpcloud** [comment]
Updated with a test, run looks good:

```
…/ibis on  fix-double-create-temp-table is 📦 v5.1.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env)
❯ pytest -m snowflake -n auto --dist loadgroup --snapshot-update --allow-snapshot-deletion -q
bringing up nodes...
........s....x...x........x.x.............x......x.....x..........xxxx...x...xx.xx......x....x.x.xxxxx....x.xxxxxxx.xxx.......x......x......xxx..x...x..xx....x..x......x.....xxxxx.x.....x..x [ 20%]
xx.x..xx.x.x..x.xx.....xx......x...x........x.......................................x..x.x....xx.x..xx.......x.x.................x.....xx....x........x.x.x.x..xxx..........xx...x...x.....x.x [ 41%]
..xx.....x.x..........x...xx.x....x.......x..x..xx..xxxx........x.xx...x.......x.x...x.....x............................x..........x....................x..................................... [ 62%]
..............x...x..................x.s.......xxx...xxs.s...x...x.....x....................x..xx...................x.....x............................................................x..x... [ 83%]
...........................................x.........x..............x..x..s.....x..........x......x..x..............x......s..x..x..s...............                                           [100%]
747 passed, 7 skipped, 154 xfailed in 113.23s (0:01:53)
```

File: ibis/backends/snowflake/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/snowflake/__init__.py]: Done! Added a test too!

**Comment by cpcloud** [comment]
Actually, I'm going to move all the work under the conditional so that we can avoid doing any expensive at all if the table already exists.

**Comment by cpcloud** [comment]
Moved nearly everything under the conditional. Tests looking good:

```
…/ibis on  fix-double-create-temp-table is 📦 v5.1.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env) took 14s
❯ pytest -m snowflake -n auto --dist loadgroup --snapshot-update --allow-snapshot-deletion -q
bringing up nodes...
s...s.x......s.........x...x..........x....x...............s..........x.............x.................x.x............x........x....x..x..x..xx....x....xx.xx..xx...x............x.x....x...x.x [ 20%]
.....x..............x..x..x....xx......x...x........................xx....x....x...........x................x....x........x..................x...........................................x.... [ 41%]
...............................x..x...x.................x...........x....x..x.....x.........x.........x.x......x...x.xx.x........xx...x.....xx.....x.......x.x...x........x.xx..xxx...x.x..... [ 62%]
x.x.xx..x...............xx.sx..xxx.x......x..............x....................................xx....x......xx.......x...........x......x...xx...x........x.....xx............................x [ 83%]
...x.....x....x..............x..xx....x..x...xx...x.xxx........x.......x...x.x.x.xxxxxxxxxxxxxxs.x.x...x.x.x.x....x..x..sx.x.x...................x..                                           [100%]
747 passed, 7 skipped, 154 xfailed in 114.20s (0:01:54)
```

**Pull request merged**